include/pmacc/algorithms/math/defines/pi.hpp

```diff
@@ -0,0 +1,47 @@
+/* Copyright 2018 Sergei Bastrakov
+ *
+ * This file is part of PMacc.
+ *
+ * PMacc is free software: you can redistribute it and/or modify
+ * it under the terms of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PMacc is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with PMacc.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+namespace pmacc
+{
+namespace algorithms
+{
+namespace math
+{
+namespace pi
+{
+
+    /** Values of pi and related constants as T_Type
+     */
+    template< typename T_Type >
+    struct Pi
+    {
+        static constexpr T_Type value = static_cast<T_Type>(3.141592653589793238462643383279502884197169399);
+        static constexpr T_Type halfValue = value / static_cast<T_Type>(2.0);
+        static constexpr T_Type quarterValue = value / static_cast<T_Type>(4.0);
```

Comment:
We should discuss the pro/con of this pre-defines   offline with @ax3l.

Reply:
Ok, I'm also not 100% sure these should exist, just added as they are part of the "M_PI package" and were used already in the code.

Reply:
Since we have `constexpr` in modern C++ we might not need those good ol' C-defines ;)

Anway, if you want to define a few, I insist we add 2 pi, aka [tau](https://tauday.com/) since it's the only true circle constant, superior to all other fractions of it.